lib/archethic_web/router.ex

```diff
@@ -25,6 +25,8 @@ defmodule ArchEthicWeb.Router do
     get("/", RootController, :index)
     get("/up", UpController, :up)
     get("/metrics", MetricsController, :index)
+    get("/faucet", FaucetController, :index)
```

Comment:
For those endpoints, you should add a condition when to display them.
We do not want to have this one production.
For example, you could add a configuration for this at compile time

Reply:
Which approach will be a wiser decision? To make a separate testnet git branch or use environment variable to detect the testnet config?

Reply:
I think the best can be to use environment variable so in dev you will have access to the faucet.

Reply:
Is it possible to get the public key from the seed/address? Without the public key I can't determine the curve.

If I can get the curve I can use this :
```elixir
new(type, data, seed, index, curve \\ Crypto.default_curve())
```

If I can get the private key I can use this instead : 
```elixir
new_with_keys(type, data, previous_private_key, previous_public_key, next_public_key)
```

But without either of them I can't generate a new transaction.

Reply:
In release `Mix.env()` doesn't work as Mix is not available. I want to inject these two lines at only compile time according to Mix.env() condition (:dev or :prod with testnet).
```elixir
get("/faucet", FaucetController, :index)
post("/faucet", FaucetController, :create_transfer)
```
Do I need to leverage Macros to inject something at compile time according to conditions?

Reply:
Using config and not Macros code injection to resolve this issue.

Reply:
Okay `Mix.env() == :dev` was enough for this since in release we can check `System.get_env("ARCHETHIC_NETWORK_TYPE") == "testnet"`.